splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/DDLConstantOperation.java

```diff
@@ -159,8 +159,11 @@ protected void storeConstraintDependenciesOnPrivileges(Activation activation,
 		DataDictionary dd = lcc.getDataDictionary();
 		DependencyManager dm = dd.getDependencyManager();
         String currentUser = lcc.getCurrentUserId(activation);
-		SettableBoolean roleDepAdded = new SettableBoolean();
-        if (! currentUser.equals( dd.getAuthorizationDatabaseOwner()) ) {
+        List<String> dependedRoleList = new ArrayList<>();
+		List<String> groupuserList = lcc.getCurrentGroupUser(activation);
+		String dbo = dd.getAuthorizationDatabaseOwner();
+        if (! currentUser.equals(dbo)
+				&& !(groupuserList != null && groupuserList.contains(dbo))) {
```

Comment:
@yxia92 What is this groupuserList check trying to do?  Do all users in the group user list automatically have permissions to eachother's objects?

Reply:
Hi @msirek , groupUserList is added as part of DB-6889. Basically, an (LDAP) user could belong to multiple groups, and the user will inherit the privileges granted to those groups. Some groups have admin privileges, like a database owner. If one of the groups is a dbo, we treat this user also as a dbo. DB-6889 adds logic to check groupusers(an example is in GenericAuthorizer.authorize()) in addition to the check of current user, but it misses the case in this function. I spot it, so just added it. It is not really related to DB-7603. thanks.